### PR TITLE
docs: adding map block

### PR DESCRIPTION
**Pull request by AnnuCode**

### CODE REVIEW

File: cli/reference/blocks/map.md [hunk 1]
```diff
@@ -4,3 +4,128 @@ description: Learn how to define complex maps and objects inside globals and let
 ---
 
 # The `map` block
+
+Use the `map` block to convert a list into a map (key-value pair) or a nested map within a `globals` or `lets` block. 
+If you have multiple objects in a list with one or more identical keys, you can merge values corresponding to the identical keys while iterating over them.
+For each object in the list, you can access the previous or the current value of any of its keys.
+In the `for-each` iteration, the iterator (default name `element`) is an object with the following properties:
+```hcl
```
Comment (soerenmartius) [inline cli/reference/blocks/map.md:12]:
```suggestion

```hcl
```

there should be a line break in between

File: cli/reference/blocks/map.md [hunk 2]
```diff
@@ -4,3 +4,128 @@ description: Learn how to define complex maps and objects inside globals and let
 ---
 
 # The `map` block
+
+Use the `map` block to convert a list into a map (key-value pair) or a nested map within a `globals` or `lets` block.
```
Comment (soerenmartius) [inline cli/reference/blocks/map.md:8]: Can we explain the actual use case here too? What's a typical use case for the `map` block?

File: cli/reference/blocks/map.md [hunk 3]
```diff
@@ -4,3 +4,128 @@ description: Learn how to define complex maps and objects inside globals and let
...
+    { name = "Trinity", product = "cape", price = 82.30 },
+    { name = "Trinity", product = "necklace", price = 25.0 },
+    { name = "Trinity", product = "sunglass", price = 100.5 },
+    { name = "Anderson", product = "ollydbg", price = 30 },
+    { name = "Morpheus", product = "boot", price = 65 },
+    { name = "Anderson", product = "cape", price = 82.30 },
+    { name = "Morpheus", product = "sunglass", price = 145.50 },
+  ]
+}
+```
```
Comment (soerenmartius) [inline cli/reference/blocks/map.md:63]:
```suggestion
```

```

File: cli/reference/blocks/map.md [hunk 4]
```diff
@@ -4,3 +4,128 @@ description: Learn how to define complex maps and objects inside globals and let
...
+
+        value {
+          total = tm_try(per_product.old.total, 0) + per_product.new.price
+        }
+      }
+    }
+  }
+}
+```
+It will result in the below object:
```
Comment (soerenmartius) [inline cli/reference/blocks/map.md:91]:
```suggestion

It will result in the below object: 

```

**Review by soerenmartius — COMMENTED** [review]
Really good job @AnnuCode, left some minor comments!

File: cli/reference/blocks/map.md (hunk 3 above)
Comment (AnnuCode) [reply cli/reference/blocks/map.md:64]: addressed

File: cli/reference/blocks/map.md (hunk 4 above)
Comment (AnnuCode) [reply cli/reference/blocks/map.md:95]: addressed

**Review by soerenmartius — APPROVED** [review]
lgtm

**Pull request merged**